ui/src/app/shared/components/object-parser.ts

```diff
@@ -10,3 +10,7 @@ export function parse<T>(value: string): T {
 export function stringify<T>(value: T, type: string) {
     return type === 'yaml' ? jsyaml.dump(value, {noRefs: true}) : JSON.stringify(value, null, '  ');
 }
+
+export function isEqual<T>(value1: T, value2: T) {
```

Comment:
I could replace this with [fast-deep-equal](https://www.npmjs.com/package/fast-deep-equal), which is already a transitive dependency: https://github.com/argoproj/argo-workflows/blob/8065edb34df0e188fd88181b4660c705997e6dcf/ui/yarn.lock#L4654

but `docs/CONTRIBUTING.md` says there has to be a "strong use case" to add a new dependency, and I'm not sure this qualifies.

Reply:
Nice work cross-checking the dependencies policy! `fast-deep-equal` hasn't had a release in 4 years and seems abandoned per https://github.com/epoberezkin/fast-deep-equal/issues/145 so I think there's other reasons to not rely on it directly as well.

A one-liner util is probably good enough as-is, though I'd add a JSDoc comment mentioning it uses `JSON.stringify` as a caveat so that it is visible in IDEs during usage.

Also I don't think this specifically belongs in `object-parser.ts`, since this file is all YAML helper functions that are primarily used in `object-editor.tsx`

Reply:
Oh, each details page does transitively use `object-editor`, so maybe this makes sense as a location actually 🤔

Reply:
That makes sense. I added a comment in 2747d1755c11b49749fb6780bcb2a46f2f4b81e4. 

You're right that this file is transitively included by each component, and I couldn't think of a better place to put this.